test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -118,14 +120,44 @@ memLimit argument is in megabytes, memLimit is a maximum amount of memory that c
 cpuLimit argument is in millicores, cpuLimit is a maximum amount of cpu that can be consumed by a single pod
 */
 func newResourceConsumer(name, nsName string, kind schema.GroupVersionKind, replicas, initCPUTotal, initMemoryTotal, initCustomMetric, consumptionTimeInSeconds, requestSizeInMillicores,
-	requestSizeInMegabytes int, requestSizeCustomMetric int, cpuLimit, memLimit int64, clientset clientset.Interface, scaleClient scaleclient.ScalesGetter, podAnnotations, serviceAnnotations map[string]string) *ResourceConsumer {
+	requestSizeInMegabytes int, requestSizeCustomMetric int, cpuLimit, memLimit int64, clientset clientset.Interface, scaleClient scaleclient.ScalesGetter, podAnnotations, serviceAnnotations map[string]string, runSidecar, busySidecar bool) *ResourceConsumer {
```

Comment:
Booleans in a long function signature are really hard to understand.  E.g. later on you add `false, false` to the end of a signature and it's not clear what that means.

Instead you could define a new type whose values would have more meaning.  E.g.

```golang
type runSidecarType bool
var (
    runSidecar   runSidecarType = true
    noRunSidecar runSidecarType = false
)

type busySidecarType bool
var (
    busy busySidecarType = true
    idle busySidecarType = false
)
```

Under the hood the data is exactly the same.  But the type system will now /1/ document the code and /2/ prevent you from accidentally putting the wrong thing in this parameter.  E.g.

```golang
    f.ScalesGetter,
    noRunSidecar,
    idle)
```

instead of

```golang
    f.ScalesGetter,
    false,
    false)
```

Reply:
Updated them to use type for those arguments